### PR TITLE
feat: Added new currency input

### PR DESCRIPTION
**Pull request by daviroo**

### CODE REVIEW

**Comment by daviroo** [comment]
After speaking with @UncleSamtoshi we are going to go in a slightly different direction with this.  We will look to replicate the logic from galoy-pay which uses a reducer to control the value displayed in the currency input.  Have moved to draft for now so I can make some changes.

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

File: app/hooks/use-price-conversion.ts [hunk 1]
```diff
@@ -68,9 +74,50 @@ export const usePriceConversion = () => {
     [priceData],
   )
 
+  const convertDisplayAmount = React.useCallback(
```
[the remark this answers is not in the document]
Comment (daviroo) [reply app/hooks/use-price-conversion.ts:77]: I think we would want to keep these separate as we discussed on the call last week.  The main point is we want to have a way of defining a `primaryCurrency` for a payment flow (which could have a `DisplayCurrency` or a `WalletCurrency`) so we can offer a recalculation of the amount at some point in the flow (if the price changes).

File: app/components/atomic/galoy-icon/galoy-icon.tsx [hunk 2]
```diff
@@ -84,6 +86,7 @@ const icons = {
   "user": User,
   "video": Video,
   "warning": Warning,
+  "switch-currency": SwitchCurrency,
```
[the remark this answers is not in the document]
Comment (nicolasburtey) [reply app/components/atomic/galoy-icon/galoy-icon.tsx:89]: Should it be the same design?

File: app/types/amounts.ts [hunk 3]
```diff
@@ -15,6 +15,8 @@ export type PaymentAmount<T extends WalletCurrency> = {
 export type DisplayAmount<T extends DisplayCurrency> = {
   amount: number
   currency: T
+  display?: string
+  precision?: number
```
[the remark this answers is not in the document]
Comment (daviroo) [reply app/types/amounts.ts]: The display values have been removed as we discussed.

File: app/components/currency-input/currency-input-state.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  ADD_DIGIT: "add-digit",
+  DELETE_DIGIT: "delete-digit",
+  CLEAR_INPUT: "clear-input",
+  TOGGLE_PRIMARY_CURRENCY: "toggle-primary-currency",
+  UPDATE_PRIMARY_DISPLAY_VALUE: "update-primary-display-value",
+  UPDATE_SECONDARY_AMOUNT: "update-secondary-amount",
+}
+
+export type ACTION_TYPE = {
+  type: string
```
[the remark this answers is not in the document]
Comment (daviroo) [reply app/components/currency-input/currency-input-state.tsx]: This is done now.

**Pull request closed without merging**